silver_braintree/payment_processors.py

```diff
@@ -370,9 +373,59 @@ def execute_transaction(self, transaction):
 
         return self._charge_transaction(transaction)
 
+    def find_lost_transaction_id(self, transaction):
+        if transaction.data.get('braintree_id'):
+            return True
+
+        search_result = braintree.Transaction.search(
+            braintree.TransactionSearch.amount.is_equal(transaction.amount),
+            braintree.TransactionSearch.payment_method_token.is_equal(
+                transaction.payment_method.token
+            ),
+            braintree.TransactionSearch.created_at.between(
+                dateutil.parser.parse(transaction.data['requested_at']) - timedelta(seconds=1),
+                transaction.created_at + timedelta(seconds=61)
+            )
+        )
+
+        transactions = list(search_result.items)
+
+        def sync_transaction_id(unsynced_transaction, transaction_list):
```

Comment:
Maybe we can extract them into more concrete representations...

Refactoring code blocks into functions and using them as normal code blocks doesn't add value.
Instead, another indirection level is added.

Maybe this example will look more readable to you:
 ```python
untracked_transactions = list(search_result.items)
if len(untracked_transactions) > 1:
     # get rid of transactions that are already tracked before trying to find a match

      untracked_transactions = [
           transaction for transaction in untracked_transactions
           if not Transaction.objects.filter(external_reference=transaction.id).exists()
       ]

if len(untracked_transactions) == 1:
    unsynced_transaction.data['braintree_id'] = untracked_transactions[0].id
    unsynced_transaction.external_reference = untracked_transactions[0].id
elif len(untracked_transactions) < 1:
    unsynced_transaction.fail(
        fail_reason="The transaction request didn't reach Braintree."
    )

unsynced_transaction.save()

return len(transaction_list) == 1